### PR TITLE
feat: beyond progressive web apps part 2

### PR DESCRIPTION
**Pull request by janl**
part two of our three part series on data sync beyond progressive web apps.

I’d love to publish this on Wednesday June 15 11am CEST.

This is already copy-edited, but would love your feedback on everything <3

[Preview](https://github.com/hoodiehq/hood.ie/blob/beyond-pwas-2/_posts/2016-06-15-beyond-progressive-web-apps-part-2.md) 👀

### CODE REVIEW

**Comment by gr2m** [comment]
`using "3", "3*", and "3**" to refer` looses the * because markup

Otherwise LGTM

**Comment by janl** [comment]
@gr2m good catch, will fix, thank you!

**Comment by jennwrites** [comment]
LGTM

**Comment by janl** [comment]
last push fixes the *-issue

File: _posts/2016-06-15-beyond-progressive-web-apps-part-2.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+title: 'Beyond Progressive Web Apps Part 2: We Need to Know What’s New'
+author: Jan
+layout: blog-post
+comments: true
+
+---
+
+This is part two of a three part series about data sync beyond Progressive Web Apps. [Please read part one, before continuing](http://hood.ie/blog/beyond-progressive-web-apps-part-1.html "Beyond Progressive Web Apps Part 1").
+
+Say we have a news app that runs on mobile devices and a server that publishes new stories. Blogs and [RSS](https://en.wikipedia.org/wiki/RSS) might be a good real-world example.
```
Comment (varjmes) [inline _posts/2016-06-15-beyond-progressive-web-apps-part-2.md]: change 'might be' to 'are'. makes you sound more confident in what you're saying (and I know you are)

Comment (janl) [inline _posts/2016-06-15-beyond-progressive-web-apps-part-2.md]: good catch!

File: _posts/2016-06-15-beyond-progressive-web-apps-part-2.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+The scenario is this: our app starts for the first time and there are no stories available for the user to read on the device. So, the app asks the server to send the latest set of stories to the device. End result:  our users get to read some stories.
```
Comment (varjmes) [inline _posts/2016-06-15-beyond-progressive-web-apps-part-2.md]: too many spaces after 'End Result:`

File: _posts/2016-06-15-beyond-progressive-web-apps-part-2.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+Later, when our apps starts for the second time, your app asks for the latest set of stories. 
+
+And here is the first interesting bit: *we don’t want to send the app the stories it already has.*
+
+Why?
+
+Four reasons:
+
+1. They already are on the device, so it's redundant
```
Comment (varjmes) [inline _posts/2016-06-15-beyond-progressive-web-apps-part-2.md]: "They are already on the device" reads better than "They already are on the device"

File: _posts/2016-06-15-beyond-progressive-web-apps-part-2.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+2. It costs us server processing time
+3. It costs our users bandwidth
+4. It adds response-time latency to the experience 
+
+As a general rule of thumb, in user experience design: don’t let the user wait unnecessarily or they will get frustrated and stop using your app, your store, whatever.
+
+So what's the solution? 
+
+We need to  request new stories from the server in a way that says “I’ve got all stories *up to this point*. Give, give me *anything that’s new*”.
```
Comment (varjmes) [inline _posts/2016-06-15-beyond-progressive-web-apps-part-2.md]: too many spaces in "to  request"

File: _posts/2016-06-15-beyond-progressive-web-apps-part-2.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+As a general rule of thumb, in user experience design: don’t let the user wait unnecessarily or they will get frustrated and stop using your app, your store, whatever.
+
+So what's the solution? 
+
+We need to  request new stories from the server in a way that says “I’ve got all stories *up to this point*. Give, give me *anything that’s new*”. 
+
+We’ll call the difference between "all stories" and "stories that exist on the client" the *delta* (i.e. difference). 
+
+The delta is what  we need to get from the server.
```
Comment (varjmes) [inline _posts/2016-06-15-beyond-progressive-web-apps-part-2.md]: Too many spaces in "what we need"

File: _posts/2016-06-15-beyond-progressive-web-apps-part-2.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+### The Index
+
+An index has two things going for it:
+
+*Order:* things will be stored in index order on disk, so reading things (e.g. stories sorted by publish date) in that order is very efficient. 
+
+*Support for ranges:* reading only part of an index from any point in the index range (e.g. all stories after a certain date) is very efficient. 
+
+Maybe the database already has a auto-increment integer as a primary key. If so, each new story gets an ID integer that is one higher the previous ID.
```
Comment (varjmes) [inline _posts/2016-06-15-beyond-progressive-web-apps-part-2.md]: "a auto-increment integer" should be "an auto-increment integer"

File: _posts/2016-06-15-beyond-progressive-web-apps-part-2.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+*Order:* things will be stored in index order on disk, so reading things (e.g. stories sorted by publish date) in that order is very efficient. 
+
+*Support for ranges:* reading only part of an index from any point in the index range (e.g. all stories after a certain date) is very efficient. 
+
+Maybe the database already has a auto-increment integer as a primary key. If so, each new story gets an ID integer that is one higher the previous ID. 
+
+If this is the case, the server is already prepared and the app only needs to store the highest ID it gets from the server in the initial request. The app can then send that ID as the high watermark for subsequent requests. 
+
+When the app receives the new batch of stories,  it stores the new highest ID as the next high watermark, and so on.
```
Comment (varjmes) [inline _posts/2016-06-15-beyond-progressive-web-apps-part-2.md]: too many spaces in ",  it"

**Comment by varjmes** [comment]
Added a few comments.

Last: get rid of your `Alt` under each of the image. The purpose of an alt tag is to explain what is going on the image, and these don't. You're doing that in the paragraph underneath the image :)

**Comment by janl** [comment]
Thank you @Charlotteis! I’ve addressed all comments in the latest push

**Comment by varjmes** [comment]
LGTM 💃

**Comment by janl** [comment]
woot published, thanks everyone!

**Pull request merged**